src/biotite/database/uniprot/download.py

```diff
@@ -0,0 +1,190 @@
+        else:
+            # 'file = None' -> store content in a file-like object
+            file = None
+        if file is None \
+                or not isfile(file) \
+                or getsize(file) == 0 \
+                or overwrite:
+            if format == "fasta":
+                r = requests.get(fetch_url + db_name + "/" + id + ".fasta")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "gff":
+                r = requests.get(fetch_url + db_name + "/" + id + ".gff")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "txt":
+                r = requests.get(fetch_url + db_name + "/" + id + ".txt")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "xml":
+                r = requests.get(fetch_url + db_name + "/" + id + ".xml")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "rdf":
+                r = requests.get(fetch_url + db_name + "/" + id + ".rdf")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "tab":
+                r = requests.get(fetch_url + db_name + "/" + id + ".tab")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            else:
+                raise ValueError(f"Format '{format}' is not supported")
+            if file is None:
+                file = io.StringIO(content)
+            else:
+                with open(file, "w+") as f:
+                    f.write(content)
+        files.append(file)
+    if verbose:
+        print("\nDone")
+    # If input was a single ID, return only a single path
+    if single_element:
+        return files[0]
+    else:
+        return files
+
+
+def _assert_valid_file(response_status_code, uid, format):
+    """
+    Checks whether the response is an actual file
+    or not.
+
+    Parameters
+    ----------
+    response_status_code: int
+        Status code of request.get.
+    uid:
+        ID of the file to be downloaded.
+    format:
+        Format of the file to be downloaded.
+    """
+    if response_status_code == 400:
+        raise RequestError("Bad request. There is a problem with your input {:}.".format(uid + "." + format))
+    elif response_status_code == 404:
+        raise RequestError("Not found. The resource {:} you requested doesn't exist.".format(uid + "." + format))
+    elif response_status_code == 410:
+        raise RequestError("Gone. The resource {:} you requested was removed.".format(uid + "." + format))
+    elif response_status_code == 500:
+        raise RequestError(
+            "Internal server error. Most likely a temporary problem, but if the problem persists please contact us.")
```

Comment:
In my opinion *us* sounds like you refer to the Biotite developers. Maybe you could use *UniProt team* instead?